src/telemetry.rs

```diff
@@ -34,3 +35,15 @@ pub fn init_subscriber(subscriber: impl Subscriber + Send + Sync) {
     LogTracer::init().expect("Failed to set logger");
     set_global_default(subscriber).expect("Failed to set subscriber");
 }
+
+/// Spawn a new blocking thread with the current span
+/// Just copied trait bounds and signature from `spawn_blocking`
+/// Remember: this is used when we have a CPU-intensive workload to perform
+pub fn spawn_blocking_with_tracing<F, R>(f: F) -> JoinHandle<R>
```

Comment:
This is a helper function to remove boilerplate from other code. It lets us spawn a tokio blocking thread within the same tracing span